src/components/VariantSelector/markup.ts

```diff
@@ -18,21 +18,32 @@ export function generateVariantSelectorHTML(_element: VariantSelector, product:
 
   return html`
     <div class="selector" part="selector">
-      ${product.options.map(option => generateOptionRowHTML(option))}<slot></slot>
+      ${product.options.map(option => generateOptionRowHTML(option, element.maxValues))}<slot></slot>
     </div>
   `
 }
 
-function generateOptionRowHTML(option: ShopifyOption) {
+function generateOptionRowHTML(option: ShopifyOption, maxValues?: number) {
   if (option.optionValues.length <= 1) {
     return ""
   }
 
+  const valuesToRender = maxValues && maxValues > 0 ? option.optionValues.slice(0, maxValues) : option.optionValues
+  const hasMore = maxValues && maxValues > 0 && option.optionValues.length > maxValues
```

Comment:
simplify to:
const hasMore = maxValues && option.optionVales.length > maxValues